### PR TITLE
Add missing properties to `ListTileTheme.merge`

### LINKED ISSUE #121974: Add `mouseCursor` and `visualDensity` arguments to `ListTileTheme.merge`
## Use case

`ListTileTheme.merge` is missing `mouseCursor` and `visualDensity` arguments and it does not query the values from the parent theme either. Consequently, `ListTileThemeData.mouseCursor` and `ListTileThemeData.visualDensity` are lost anywhere `ListTileTheme.merge` is used:

```sh
$ git grep ListTileTheme.merge
lib/src/material/drawer.dart:    return ListTileTheme.merge(
lib/src/material/expansion_tile.dart:          ListTileTheme.merge(
```

## Test case

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      theme: ThemeData(
        listTileTheme: ListTileThemeData(
          mouseCursor: MaterialStateProperty.all(SystemMouseCursors.forbidden),
          visualDensity: VisualDensity.compact,
        ),
      ),
      home: Scaffold(
        body: Column(
          children: [
            ListTile(
              title: const Text('ListTile'),
              subtitle: const Text('forbidden, compact'),
              onTap: () {},
            ),
            ListTileTheme.merge(
              // mouseCursor
              // visualDensity
              child: ListTile(
                title: const Text('ListTile'),
                subtitle: const Text('click, null'),
                onTap: () {},
              ),
            ),
          ],
        ),
      ),
    ),
  );
}
```

[list_tile_theme_merge.webm](https://user-images.githubusercontent.com/140617/222974670-a3b46009-dd65-412a-bc89-9bae4cae6a9f.webm)

## Proposal

Add `mouseCursor` and `visualDensity` arguments to `ListTileTheme.merge` and specify them for the newly created `ListTileThemeData`, respectively.

### PR DESCRIPTION
**Pull request by jpnurmi**
This PR adds missing `mouseCursor` and `visualDensity` arguments to `ListTileTheme.merge`, makes it query the respective values from the parent theme, and passes them to the newly created `ListTileThemeData`. This ensures that `ListTileThemeData.mouseCursor` and `ListTileThemeData.visualDensity` are not lost when using `ListTileTheme.merge`.

Fixes: #121974

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
Thanks for the PR! @jpnurmi 

This looks good, however, we're just testing explicit values in the `ListTileTheme.merge`. 

When trying something like this, parent values are null in your test. 
```dart
        print('mouseCursor: $mouseCursor, parent.mouseCursor: ${parent.mouseCursor}');
        print('mouseCursor: $visualDensity: parent.visualDensity: ${parent.visualDensity}');
```

It would be great if the test checks that all properties actually getting merged with parent `ListTileTheme`, this will ensure all properties are supported as you're testing them here.

Looks like we're also missing `titleTextStyle`, `subtitleTextStyle`, and `leadingAndTrailingTextStyle`. I added these properties and somehow missed the `ListTileTheme.merge`.

File: packages/flutter/lib/src/material/list_tile_theme.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/list_tile_theme.dart:492]: Please add missing `titleTextStyle`, `subtitleTextStyle`, and `leadingAndTrailingTextStyle`.

```dart
    Color? textColor,
    TextStyle? titleTextStyle,
    TextStyle? subtitleTextStyle,
    TextStyle? leadingAndTrailingTextStyle,
    EdgeInsetsGeometry? contentPadding,
    Color? tileColor,
    Color? selectedTileColor,
    bool? enableFeedback,
    double? horizontalTitleGap,
    double? minVerticalPadding,
    double? minLeadingWidth,
    ListTileTitleAlignment? titleAlignment,
    MaterialStateProperty<MouseCursor?>? mouseCursor,
    VisualDensity? visualDensity,
    required Widget child,

```

File: packages/flutter/lib/src/material/list_tile_theme.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/list_tile_theme.dart:516]:
```dart
            textColor: textColor ?? parent.textColor,
            titleTextStyle: titleTextStyle ?? parent.titleTextStyle,
            subtitleTextStyle: subtitleTextStyle ?? parent.subtitleTextStyle,
            leadingAndTrailingTextStyle: leadingAndTrailingTextStyle ?? parent.leadingAndTrailingTextStyle,
            contentPadding: contentPadding ?? parent.contentPadding,
            tileColor: tileColor ?? parent.tileColor,
            selectedTileColor: selectedTileColor ?? parent.selectedTileColor,
            enableFeedback: enableFeedback ?? parent.enableFeedback,
            horizontalTitleGap: horizontalTitleGap ?? parent.horizontalTitleGap,
            minVerticalPadding: minVerticalPadding ?? parent.minVerticalPadding,
            minLeadingWidth: minLeadingWidth ?? parent.minLeadingWidth,
            titleAlignment: titleAlignment ?? parent.titleAlignment,
            mouseCursor: mouseCursor ?? parent.mouseCursor,
            visualDensity: visualDensity ?? parent.visualDensity,
```

File: packages/flutter/test/material/list_tile_theme_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/list_tile_theme_test.dart:867]:
```suggestion

  testWidgets('ListTileTheme.merge supports all properties', (WidgetTester tester) async {
    Widget buildFrame() {
      return MaterialApp(
        theme: ThemeData(
          listTileTheme: const ListTileThemeData(
            dense: true,
            shape: StadiumBorder(),
            style: ListTileStyle.drawer,
            selectedColor: Color(0x00000001),
            iconColor: Color(0x00000002),
            textColor: Color(0x00000003),
            titleTextStyle: TextStyle(color: Color(0x00000004)),
            subtitleTextStyle: TextStyle(color: Color(0x00000005)),
            leadingAndTrailingTextStyle: TextStyle(color: Color(0x00000006)),
            contentPadding: EdgeInsets.all(100),
            tileColor: Color(0x00000007),
            selectedTileColor: Color(0x00000008),
            horizontalTitleGap: 200,
            minVerticalPadding: 300,
            minLeadingWidth: 400,
            enableFeedback: true,
            titleAlignment: ListTileTitleAlignment.bottom,
            mouseCursor: MaterialStateMouseCursor.textable,
            visualDensity: VisualDensity.comfortable,
          ),
        ),
        home: Material(
          child: Center(
            child: Builder(
              builder: (BuildContext context) {
                return ListTileTheme.merge(
                  dense: false,
                  shape: const RoundedRectangleBorder(),
                  style: ListTileStyle.list,
                  selectedColor: const Color(0x00000009),
                  iconColor: const Color(0x0000000A),
                  textColor: const Color(0x0000000B),
                  titleTextStyle: const TextStyle(color: Color(0x0000000C)),
                  subtitleTextStyle: const TextStyle(color: Color(0x0000000D)),
                  leadingAndTrailingTextStyle: const TextStyle(color: Color(0x0000000E)),
                  contentPadding: const EdgeInsets.all(500),
                  tileColor: const Color(0x0000000F),
                  selectedTileColor: const Color(0x00000010),
                  horizontalTitleGap: 600,
                  minVerticalPadding: 700,
                  minLeadingWidth: 800,
                  enableFeedback: false,
                  titleAlignment: ListTileTitleAlignment.top,
                  mouseCursor: MaterialStateMouseCursor.clickable,
                  visualDensity: VisualDensity.compact,
                  child: const ListTile(),
                );
              }
            ),
          ),
        ),
      );
    }

    await tester.pumpWidget(buildFrame());
    final ListTileThemeData theme = ListTileTheme.of(tester.element(find.byType(ListTile)));
    expect(theme.dense, false);
    expect(theme.shape, const RoundedRectangleBorder());
    expect(theme.style, ListTileStyle.list);
    expect(theme.selectedColor, const Color(0x00000009));
    expect(theme.iconColor, const Color(0x0000000A));
    expect(theme.textColor, const Color(0x0000000B));
    expect(theme.titleTextStyle, const TextStyle(color: Color(0x0000000C)));
    expect(theme.subtitleTextStyle, const TextStyle(color: Color(0x0000000D)));
    expect(theme.leadingAndTrailingTextStyle, const TextStyle(color: Color(0x0000000E)));
    expect(theme.contentPadding, const EdgeInsets.all(500));
    expect(theme.tileColor, const Color(0x0000000F));
    expect(theme.selectedTileColor, const Color(0x00000010));
    expect(theme.horizontalTitleGap, 600);
    expect(theme.minVerticalPadding, 700);
    expect(theme.minLeadingWidth, 800);
    expect(theme.enableFeedback, false);
    expect(theme.titleAlignment, ListTileTitleAlignment.top);
    expect(theme.mouseCursor, MaterialStateMouseCursor.clickable);
    expect(theme.visualDensity, VisualDensity.compact);
  });
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by TahaTesser — APPROVED** [review]
LGTM! 

I need a second review from @HansMuller before merging this.

File: packages/flutter/test/material/list_tile_theme_test.dart
Comment (jpnurmi) [reply packages/flutter/test/material/list_tile_theme_test.dart]: Thanks for the help!

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**
